src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -262,6 +292,34 @@ int main(int argc, char *argv[])
       data::Save(CLI::GetParam<string>("neighbors_file"), neighbors);
     if (CLI::HasParam("distances_file"))
       data::Save(CLI::GetParam<string>("distances_file"), distances);
+
+    // Calculate the effective error, if desired.
+    if (CLI::HasParam("effective_error"))
+    {
+      const string exactFile = CLI::GetParam<string>("effective_error");
+      arma::mat distancesExact;
+      data::Load(exactFile, distancesExact, true);
+
+      if (distancesExact.n_elem != distances.n_elem)
+        Log::Fatal << "The effective error file must have the same number of "
+          << "values than the set of distances being queried!" << endl;
+
+      double effectiveError = 0;
+      size_t cases = 0;
+      for (size_t i = 0; i < distances.n_elem; i++)
+      {
+        if (distancesExact(i) != 0 && distances(i) != DBL_MAX)
+        {
+          effectiveError += (distances(i) - distancesExact(i)) /
+              distancesExact(i);
+          cases++;
+        }
+      }
+      if (cases)
+        effectiveError /= cases;
+
+      Log::Info << "Effective error: " << effectiveError << endl;
+    }
```

Comment:
Do you think it would be better to move the effective relative error calculation into the `NeighborSearch` class, perhaps as a static method?  You could pass in the two distance matrices as input and get the effective relative error as output.  That way people could reuse this functionality elsewhere.